### PR TITLE
🎉 New Source Delighted!

### LINKED ISSUE #2969: New source: Delighted by Qualtrics
## Tell us about the new integration you’d like to have
https://delighted.com/

## Describe the alternative you are considering or using

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368102460001) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by rparrapy**
## What
Add Delighted source with people, unsubscribes and bounces streams

## How
Native connector using the Python CDK

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

This closes #2969.

### CODE REVIEW

**Comment by rparrapy** [comment]
I’m trying to build a connector for Delighted using the CDK. People, unsubscribes and bounces were pretty straightforward. Survey responses are tricky, mainly because of the person_properties attribute of the response which is an object with dynamic attributes (per survey, per account).

I found the docs about dynamic schemas but I have a couple of questions:

- Can overriding  get_json_schema (in theory) be used to infer the schema from a subset of the records? i.e. could I read from the stream to dinamically build the schema for person_properties? Genson could help if so.
- Is there an example of any source that does something similar? If so, I’d appreciate if you can point me towards it.

File: airbyte-integrations/connectors/source-delighted/source_delighted/source.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    limit = 100
+
+    # Define primary key to all streams as primary key
+    primary_key = "id"
+
+    def __init__(self, since: int, api_key: str, **kwargs):
+        super().__init__(**kwargs)
+        self.since = since
+        self.api_key = api_key
+        self.since_id = 0
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-delighted/source_delighted/source.py:52]: this seems unused?

**Review by sherifnada — COMMENTED** [review]
very exciting @rparrapy ! is this ready for review?

**Comment by sherifnada** [comment]
> Can overriding  get_json_schema (in theory) be used to infer the schema from a subset of the records?

yes — you could do something like:
```
def get_json_schema():  
  records_generator = self.read_records(...)
  first_100 = [next(records_generator) for i in range(100)]
  schema = infer_schema(first_100)
  return schema
```
but I would generally recommend this as a last last resort if the schema is not declared ahead of time (either via API docs or an endpoint which describes the schema of custom objects) because schema inference can be very unreliable (doesn’t catch anyOf or oneOf very well, some fields might be null in some records and therefore they can be missed, etc..)

is there an API reference or schema-describing endpoint which can help here?


> Is there an example of any source that does something similar? If so, I’d appreciate if you can point me towards it.

I don’t think there is an example of a source which infers schema from records today, no

**Comment by tuliren** [comment]
Hi @rparrapy, any update on this PR? Do you need more help in addition to the answers provided by Sherif? No pressure if you currently don't have time to wrap it up. We can take it over if necessary.

**Comment by rparrapy** [comment]
It just hasn't been a priority for the past couple of weeks. I'd like to give it a go next week. Unfortunately, I haven't found a way to get the schema ahead of time, so inference might be needed.

**Comment by rparrapy** [comment]
Just wanted to let everyone know that I won't be able to work on this for at least 2 weeks from now. So if anyone has some buffer and wants to jump in feel free to do so.

**Comment by tuliren** [comment]
This PR still needs quite some work. It's not ready for review yet. I will remove myself from the assignee.

**Review by tuliren — COMMENTED** [review]
This PR still needs quite some work. It's not ready for review yet. I will remove myself from the assignee.

**Comment by rparrapy** [comment]
Would like to pick this up again... wondering if it could be part of the Hacktoberfest contest. Thoughts @marcosmarxm ?

**Review by marcosmarxm — APPROVED** [review]
Need some small changes I made in https://github.com/airbytehq/airbyte/pull/7540 but is working! thanks @rparrapy

**Pull request merged**